obspy/io/scardec/core.py

```diff
@@ -0,0 +1,457 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+SCARDEC file format support for ObsPy.
+These files contain source solutions calculated at IPGP using the
+SCARDEC method [Vallee2011]_. These solutions contain a focal mechanism
+and a source time function in form of a regularly sampled moment rate.
+An equivalent moment tensor is automatically calculated. Writing of SCARDEC
+files is only possible, if the event contains a moment rate function.
+
+:copyright:
+    The ObsPy Development Team (devs@obspy.org)
+:license:
+    GNU Lesser General Public License, Version 3
+    (https://www.gnu.org/copyleft/lesser.html)
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA @UnusedWildImport
+
+import uuid
+import warnings
+
+import numpy as np
+
+from obspy import UTCDateTime
+from obspy.core.event import (Catalog, Comment, Event, EventDescription,
+                              Origin, Magnitude, FocalMechanism, MomentTensor,
+                              Tensor, SourceTimeFunction, NodalPlane,
+                              NodalPlanes)
+from obspy.geodetics import FlinnEngdahl
+
+
+_fe = FlinnEngdahl()
+
+
+def _get_resource_id(scardecname, res_type, tag=None):
+    """
+    Helper function to create consistent resource ids.
+    """
+    res_id = "smi:local/scardec/%s/%s" % (scardecname, res_type)
+    if tag is not None:
+        res_id += "#" + tag
+    return res_id
+
+
+def _buffer_proxy(filename_or_buf, function, reset_fp=True,
+                  file_mode="rb", *args, **kwargs):
```

Comment:
This function is now defined for the fourth time I think. But it's not your fault really, it was duplicated before when it shouldn't have been..